protocol/daemons/types/daemon_client.go

```diff
@@ -0,0 +1,8 @@
+package types
+
+type DaemonClient interface {
```

Comment:
nit: I would only add the `DaemonClient` once it is necessary.

Also, `dydx.App` is likely to know the concrete types of the `daemons` so it won't need an interface type to call stop on so it is unlikely we would need the `DaemonClient` interface any time soon.